archaius2-archaius1-bridge/src/main/java/com/netflix/archaius/bridge/Archaius1DelegatingConfig.java

```diff
@@ -0,0 +1,74 @@
+package com.netflix.archaius.bridge;
+
+import java.util.Iterator;
+import java.util.Properties;
+
+import org.apache.commons.configuration.AbstractConfiguration;
+import org.apache.commons.configuration.event.ConfigurationEvent;
+import org.apache.commons.configuration.event.ConfigurationListener;
+
+import com.netflix.archaius.api.Config;
+import com.netflix.archaius.api.config.SettableConfig;
+import com.netflix.archaius.config.AbstractConfig;
+import com.netflix.config.ConfigurationManager;
+
+public class Archaius1DelegatingConfig extends AbstractConfig implements SettableConfig {
```

Comment:
I think going through Configmanager.getConfigInstance().setProperty is not what we want here. If I understand the use case right, then this is for Archaius2 based libraries. However, Configmanager.getConfigInstance().setProperty is 3 levels higher than library in coinfiguration world (both Archaius1 and Archaius2). Basically, this opens door for a Archaius2 based library to upset the Archaius1 hierarchy.
I'm not sure the exact right solution at this point. But I think the only thing this class has to do is to load the library properties and then grab the library sub-config from the delegate Archaius1 configuration and put the library properties in there. I think if you develop an Archaius2 based library, you're not supposed to dynamically change the properties in code. In application, yes, but not in libraries.